src/main/java/map/BattleMap.java

```diff
@@ -17,11 +17,60 @@ public BattleMap(int size) {
     this.map.addAll(Collections.nCopies(size * size, 0));
   }
 
+  public List<Integer> getMap() {
+    return map;
+  }
+
+  public Integer getSize() {
+    return size;
+  }
+
+  public List<Point> createShipsPositions(Point p, Integer length, Boolean is_Horizontal) {
+    int i = 0;
+    List<Point> list = new ArrayList<>();
+    while (i < length) {
+      if (is_Horizontal) {
+        if (p.y + i > size)
+          throw new IndexOutOfBoundsException("The ship positions that rely on a given length are out of the map size");
+        Point point1 = new Point(p.x, p.y + i);
+        list.add(point1);
+        i++;
+      } else {
+        if (p.x + i >= size)
+          throw new IndexOutOfBoundsException("The ship positions that rely on a given length are out of the map size");
+        Point point1 = new Point(p.x + i, p.y);
+        list.add(point1);
+        i++;
+      }
+    }
+    return list;
+  }
+
+  public boolean isBusy(Point point) {
+    return getAt(point) == 1;
+  }
+
+  public boolean isSunk(List<Point> list) {
+    return list.stream().allMatch(point -> getAt(point) == 2);
+  }
+
+  public boolean contains(List<Point> pointList, Point specPoint) {
+    return pointList.contains(specPoint);
+  }
+
   public void putShip(Point p) {
     map.add(p.to1D(size), 1);
   }
 
+  public void changeShipState(Point point) {
+    map.add(point.to1D(size), 2);
+  }
+
   public Integer getAt(Point p) {
     return map.get(p.to1D(size));
   }
+
+  public void printMap() {
```

Comment:
We have a domain model `BattleMap`, yet we print it in the terms of its inner Java Map. Why not overwrite the `toString()` of our model and use that instead?